test_pytest_html.py

```diff
@@ -287,12 +287,72 @@ def pytest_runtest_makereport(item, call):
                     report.extra = [extras.image('{0}')]
         """.format(content))
         testdir.makepyfile('def test_pass(): pass')
-        result, html = run(testdir)
+        result, html = run(testdir, 'report.html', '--self-contained-html')
         assert result.ret == 0
         assert '<a class="image" href="#" target="_blank">Image</a>' in html
         src = 'data:image/png;base64,{0}'.format(content)
         assert '<a href="#"><img src="{0}"/></a>'.format(src) in html
 
+    def test_extra_image_separeted(self, testdir):
+        content = b64encode(str(random.random())
+                            .encode('utf-8')).decode('ascii')
+        testdir.makeconftest("""
+            import pytest
+            @pytest.mark.hookwrapper
+            def pytest_runtest_makereport(item, call):
+                outcome = yield
+                report = outcome.get_result()
+                if report.when == 'call':
+                    from pytest_html import extras
+                    report.extra = [extras.image('{0}')]
+        """.format(content))
+        testdir.makepyfile('def test_pass(): pass')
+        result, html = run(testdir)
+        hash_key = ('test_extra_image_separeted.py::'
+                    'test_pass1-1').encode('utf-8')
+        hash_generator = hashlib.md5()
+        hash_generator.update(hash_key)
+        assert result.ret == 0
+        src = os.path.join('assets',
+                           '{0}.png'.format(hash_generator.hexdigest()))
+        a_img = ('<a class="image" href="{0}" '
+                 'target="_blank">Image</a>'.format(src))
+        assert a_img in html
+        assert '<a href="{0}"><img src="{1}"/></a>'.format(src, src) in html
```

Comment:
You can repeat `{0}` rather than pass the same variable into the formatter twice.